ts/nni_manager/core/nniTensorboardManager.ts

```diff
@@ -0,0 +1,208 @@
+        this.setTensorboardVersion();
+        this.nniManager = component.get(Manager);
+    }
+
+    public async startTensorboardTask(tensorboardParams: TensorboardParams): Promise<TensorboardTaskDetail> {
+        const trialJobIds = tensorboardParams.trials;
+        const trialJobIdList: string[] = [];
+        const trialLogDirectoryList: string[] = [];
+        await Promise.all(trialJobIds.split(',').map(async (trialJobId) => {
+            const trialDataPath = await this.nniManager.getTrialOutputLocalPath(trialJobId);
+            mkDirPSync(trialDataPath);
+            trialJobIdList.push(trialJobId);
+            trialLogDirectoryList.push(trialDataPath);
+        }));
+        this.log.info(`tensorboard: ${trialJobIdList} ${trialLogDirectoryList}`);
+        return this.startTensorboardTaskProcess(trialJobIdList, trialLogDirectoryList);
+    }
+
+    private async startTensorboardTaskProcess(trialJobIdList: string[], trialLogDirectoryList: string[]): Promise<TensorboardTaskDetail> {
+        const host = 'localhost';
+        const port = await getFreePort(host, 6006, 65535);
+        const command = this.getTensorboardStartCommand(trialJobIdList, trialLogDirectoryList, port);
+        this.log.info(`tensorboard start command: ${command}`);
+        const tensorboardProc: ChildProcess = getTunerProc(command, 'ignore', process.cwd(), process.env);
+        const tensorboardTask = new TensorboardTaskDetail(uniqueString(5), 'RUNNING', trialJobIdList, trialLogDirectoryList);
+        this.tensorboardTaskMap.set(tensorboardTask.id, tensorboardTask);
+        tensorboardTask.pid = tensorboardProc.pid;
+        tensorboardTask.port = `${port}`;
+        this.log.info(`tensorboard task id: ${tensorboardTask.id}`);
+        this.updateTensorboardTask(tensorboardTask.id);
+        return tensorboardTask;
+    }
+
+    private getTensorboardStartCommand(trialJobIdList: string[], trialLogDirectoryList: string[], port: number): string {
+        if (this.tensorboardVersion === undefined) {
+            this.setTensorboardVersion();
+            if (this.tensorboardVersion === undefined) {
+                throw new Error(`Tensorboard may not installed, if you want to use tensorboard, please check if tensorboard installed.`);
+            }
+        }
+        if (trialJobIdList.length !== trialLogDirectoryList.length) {
+            throw new Error('trial list length does not match');
+        }
+        if (trialJobIdList.length === 0) {
+            throw new Error('trial list length is 0');
+        }
+        let logdirCmd = '--logdir';
+        if (this.tensorboardVersion >= '2.0') {
+            logdirCmd = '--bind_all --logdir_spec'
+        }
+        try {
+            const logRealPaths: string[] = [];
+            trialJobIdList.forEach((trialJobId, idx) => {
+                const realPath = fs.realpathSync(trialLogDirectoryList[idx]);
+                logRealPaths.push(`${trialJobId}:${realPath}`);
```

Comment:
Can we add a trial sequence id here for readability?

Reply:
good suggestion, add it.